app/models/user.rb

```diff
@@ -177,11 +175,16 @@ def update_stats
   end
 
   # Override Devise::Model::Lockable#lock_access! to add event logging
-  def lock_access!
-    EventLog.record_event(User.find_by_email(self.email), EventLog::ACCOUNT_LOCKED)
-    super
-
-    UserMailer.locked_account_explanation(self).deliver_later
+  def lock_access! opts = {}
+    event = case locked_reason
```

Comment:
Only in that I wanted to be explicit about the cases and have nil as the default.